modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java

```diff
@@ -430,6 +430,10 @@ private void processCruise() {
             if (pcraft.getType().getCruiseOnPilot()) {
                 dy = pcraft.getType().getCruiseOnPilotVertMove();
             }
+            final int gearshift = pcraft.getCurrentGear();
+            dx *= gearshift;
+            dy *= gearshift;
+            dz *= gearshift;

```

Comment:
See above comment.  At the very least this should only work if in direct movement mode, not tick cooldown.